service/src/geopm/MSRIOGroup.hpp

```diff
@@ -112,9 +112,18 @@ namespace geopm
             /// @brief Return the JSON string for the MSR data
             ///        associated with the given cpuid.
             static std::string platform_data(int cpu_id);
+
+            enum MsrConfigWarningPreference_e {
+                SILENCE_CONFIG_DEPRECATION_WARNING,
+                EMIT_CONFIG_DEPRECATION_WARNING,
+            };
             /// @brief Returns the filenames for user-defined MSRs if
             ///        found in the plugin path.
-            static std::set<std::string> msr_data_files(void);
+            /// @param [in] emit_config_deprecation_warning Whether to emit a warning
+            ///        if MSR config files are found while searching the plugin
+            ///        path instead of while searching config file locations.
+            static std::set<std::string> msr_data_files(
+                    MsrConfigWarningPreference_e warning_preference = SILENCE_CONFIG_DEPRECATION_WARNING);
```

Comment:
An enum in a function definition 😲 I guess since its private it makes sense.  We may want to rethink our use of enum variables always being cast to integers at a future date.

Reply:
I'm not super attached to this approach. My main motivation was to try going for calls easier to interpret than `msr_data_files(true)`.